src/app/services/services.module.ts

```diff
@@ -161,11 +163,17 @@ export function initFactory(): Function {
         authService.init();
         const htmlEl = window.document.documentElement;
         htmlEl.classList.add('locale_' + i18nService.translationLocale);
-        let theme = await storageService.get<string>(ConstantsService.themeKey);
-        if (theme == null) {
-            theme = 'light';
-        }
+
+        const theme = await platformUtilsService.getEffectiveTheme();
         htmlEl.classList.add('theme_' + theme);
+        platformUtilsService.onDefaultSystemThemeChange(async sysTheme => {
```

Comment:
`index.html` only sets the initial theme, it doesn't set up the `onDefaultSystemThemeChange` handler.

Your point still applies to setting the initial theme, we could delete it from lines 167-168 in this file because it's handled in `index.html`. However, `index.html` is more brittle because it can't hook into the theme enums or the `platformUtilsService` methods, it's a bit of an anomaly. I'm not sure whether it's worth keeping this here as a backup, even though the duplication is ugly. What do you think?